src/audio/mux/mux.c

```diff
@@ -527,18 +527,24 @@ static int mux_process(struct processing_module *mod,
 	struct comp_buffer __sparse_cache *source_c;
 	struct list_item *clist;
 	const struct audio_stream __sparse_cache *sources_stream[MUX_MAX_STREAMS] = { NULL };
-	int frames;
+	int frames = 0;
 	int sink_bytes;
 	int source_bytes;
-	int i;
+	int i, j;
 
 	comp_dbg(dev, "mux_process()");
 
 	/* align source streams with their respective configurations */
+	j = 0;
 	list_for_item(clist, &dev->bsource_list) {
 		source = container_of(clist, struct comp_buffer, sink_list);
 		source_c = buffer_acquire(source);
 		if (source_c->source->state == dev->state) {
+			if (frames)
+				frames = MIN(frames, input_buffers[j].size);
+			else
+				frames = input_buffers[j].size;
```

Comment:
this is rather odd, why not initialize frame to MAX_INT and do the line 544 unconditionally?

Reply:
@plbossart This was in previous version of the patch, but bsource_list may be empty in which case we'd end up with frames==MAX_INT